MarsRover/Rover.cs

```diff
@@ -66,15 +61,15 @@ private void TurnRight()
             };
         }
         
-        private void MoveForward(int maxWidth, int maxHeight) 
+        private Location MoveForward(int maxWidth, int maxHeight) 
         {
             var x = Location.X;
             var y = Location.Y;
             var xLeft = x == 0 ? maxWidth - 1 : x - 1;
             var xRight = x == maxWidth - 1 ? 0 : x + 1;
             var yTop = y == 0 ? maxHeight - 1 : y - 1;
             var yBottom = y == maxHeight - 1 ? 0 : y + 1;
-            Location = Direction switch
+            return Location = Direction switch

```

Comment:
As @indianbread mentioned further down, we want to assign the location when we `MoveForward` or `MoveBackward`. The value of the location that we assign can be determined by a `GetTargetLocation` method as you've created. Except the logic for actually calculating that location would be the responsibility of the `GetTargetLocation` method i.e. `MoveForward` and `MoveBackward` should call `GetTargetLocation` rather than `GetTargetLocation` calling `MoveForward` or `MoveBackward`